### PR TITLE
Database migrations

### PR DESCRIPTION
**Pull request by niekcandaele**
Generated sequelize models from the output database of `create_tables.js`, these can be easily used for migrations. After all migrations for creating tables are created, these models should be removed again. Migrating all DB operations in CSMM to Sequelize is for another time :smiley: 

The migration executes a query like this:
```sql
CREATE TABLE IF NOT EXISTS `user` (`createdAt` BIGINT, `updatedAt` BIGINT, `id` INTEGER(11) NOT NULL auto_increment , `steamId` VARCHAR(255) UNIQUE, `discordId` VARCHAR(255), `username` VARCHAR(255), `avatar` VARCHAR(255), `admin` INTEGER(1), `banned` INTEGER(1), PRIMARY KEY (`id`)) ENGINE=InnoDB DEFAULT CHARSET=utf8m4 COLLATE utf8mb4_unicode_ci;
```

Because it uses `IF NOT EXISTS` I think this will not cause anything to break for existing databases but this should be extensively tested. 
Otherwise the solution is probably to manually add a record saying that the init migration has 'run' to the Sequelize meta table. This would be a major bump though so I hope that's not going to be needed

### CODE REVIEW

**Comment by halkeye** [comment]
https://github.com/halkeye/7-days-to-die-server-manager/commit/c697b196f798e7b2bff07454e61280bbfc96ed6e from a while ago actually started converting models to sequelice too

**Comment by niekcandaele** [comment]
Lol, is find -> findAll really the only API difference between Waterline and Sequelize? :P 

I see you used a sails hook for that branch, I'd like to move away from sails related stuff as much as possible. Since this is a new feature, it allows us to make it in normal JS. WDYT?

**Comment by halkeye** [comment]
```
docker-compose exec -T mysql mysqladmin -u csmm -pmysecretpasswordissosecure drop csmm -f
docker-compose exec -T mysql mysqladmin -u csmm -pmysecretpasswordissosecure create csmm
npm run db:migrate
docker-compose exec -T mysql mysqldump -d -u csmm -pmysecretpasswordissosecure csmm > ../csmm-tables-20200927-migrated.sql
```


[Uploading sqls.zip…]()

**Comment by halkeye** [comment]
```
docker-compose exec -T mysql mysqladmin -u csmm -pmysecretpasswordissosecure drop csmm -f
docker-compose exec -T mysql mysqladmin -u csmm -pmysecretpasswordissosecure create csmm
docker-compose exec -T mysql mysqldump -d -u csmm -pmysecretpasswordissosecure csmm < ../csmm-us.sql
npm run db:migrate
docker-compose exec -T mysql mysqldump -d -u csmm -pmysecretpasswordissosecure csmm > ../csmm-tables-20200927-existing-migrated.sql
```

[sqls.zip](https://github.com/CatalysmsServerManager/7-days-to-die-server-manager/files/5290159/sqls.zip)

the us db looks good

File: migrations/20200927000000-add-indexes.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+module.exports = {
+  up: async (queryInterface, Sequelize) => {
+    const {DataTypes} = Sequelize;
+    const transaction = await queryInterface.sequelize.transaction();
```
Comment (niekcandaele) [inline migrations/20200927000000-add-indexes.js:6]: Does this work? I thought you have to pass the transaction object to queries

```js
      queryInterface.changeColumn('gblcomment', 'content', {
        type: DataTypes.TEXT,
        allowNull: true
      }, { transaction });
```

Something like that?

Comment (halkeye) [reply migrations/20200927000000-add-indexes.js:6]: I honestly don't know I just followed stuff I found online.

A migration should use the same connection so in theory it should work

**Pull request merged**